### DISCUSSION TITLE
delete collection and how to recover

### DISCUSSION THREAD

**Discussion by ECHO2008** [milvus-io/milvus#34899] [Q&A and General discussion]
If the drop_collection operation is performed in the code, is there a way to recover the deleted collection or data

**Comment by xiaofan-luan** [comment]
if you have backup, you can recover it from backup.
Otherwise, if is is less than 24 hours, you should be abled to find it on S3 or MInio
simply cp the directory on minio. you need to know your collection ID(Or you can simply backup the entire directory

**Reply by ECHO2008** [reply]
I am using minio. Is there any detailed recovery procedure or related documentation

**Reply by xiaofan-luan** [reply]
please try to check the minio，see if the directory is still there.

**Reply by xiaofan-luan** [reply]
you need copy minio dir to a seperate dir, and manully read the file out and reinsert into the milvus cluster.
Spark connector can probably help you on it
